src/core/Table/hooks/useColumnVisibility.tsx

```diff
@@ -2,72 +2,31 @@
-import React from 'react';
-import { ColumnInstance, HeaderProps, Hooks } from 'react-table';
-import { Checkbox } from '../../Checkbox';
-import SvgColumnManager from '@itwin/itwinui-icons-react/cjs/icons/ColumnManager';
-import { DropdownMenu } from '../../DropdownMenu';
-import { IconButton } from '../../Buttons/IconButton';
-import { MenuItem } from '../../Menu';
-
-export const COLUMN_MANAGER_ID = 'iui-table-column-manager';
-
+import { ColumnInstance, Hooks } from 'react-table';
+import { COLUMN_MANAGER_ID, ManagerColumn } from '../columns';
 export const useColumnVisibility = <T extends Record<string, unknown>>(
   hasColumnManager: boolean,
 ) => (hooks: Hooks<T>) => {
   if (!hasColumnManager) {
     return;
   }
 
-  hooks.allColumns.push((columns: ColumnInstance<T>[]) => [
-    ...columns,
-    {
-      id: COLUMN_MANAGER_ID,
-      minWidth: 48,
-      width: 48,
-      maxWidth: 48,
-      disableReordering: true,
-      columnClassName: 'iui-slot',
-      cellClassName: 'iui-slot',
-      Header: ({ allColumns }: HeaderProps<T>) => {
-        const [isOpen, setIsOpen] = React.useState(false);
-        const headerCheckBoxes = () =>
-          allColumns
-            //Filters out any default columns made such as selection and expansion
-            .filter(({ id }) => !id.includes('iui-table'))
-            .map((column) => {
-              const { onChange, checked } = column.getToggleHiddenProps();
-              return (
-                <MenuItem
-                  key={column.id}
-                  icon={
-                    <Checkbox
-                      id={`iui-column-${column.id}`}
-                      checked={checked}
-                      disabled={column.disableToggleVisibility}
-                      onClick={(e) => e.stopPropagation()}
-                      onChange={onChange}
-                    />
-                  }
-                  onClick={() => column.toggleHidden(checked)}
-                  disabled={column.disableToggleVisibility}
-                >
-                  {column.Header}
-                </MenuItem>
-              );
-            });
-        return (
-          <DropdownMenu
-            menuItems={headerCheckBoxes}
-            onHide={() => setIsOpen(false)}
-            onShow={() => setIsOpen(true)}
-          >
-            <IconButton styleType='borderless' isActive={isOpen}>
-              <SvgColumnManager />
-            </IconButton>
-          </DropdownMenu>
-        );
-      },
-    },
-  ]);
+  hooks.allColumns.push((columns: ColumnInstance<T>[]) => {
+    const hasManagerColumn = columns.find((c) => c.id === COLUMN_MANAGER_ID);
+    if (hasManagerColumn) {
+      return columns;
+    }
+    const lastColumn = columns[columns.length - 1];
+    const managerColumn = ManagerColumn();
+    // If the last column is an iui-slot that has it's own Cell props, don't generate a separate column for Column Manager
+    if (
+      lastColumn.cellClassName?.includes('iui-slot') &&
```

Comment:
So I was thinking a little bit different... Users by themselves do not add `iui-slot` class.. They do not know about it and we do not document it anywhere.
The purpose of creating this column as a component was for users to actually use that for their actionable last column (to display menu, for example). And then what would you do is to check if such column exists and just update its header part, leaving everything else.

Reply:
This has been changed. Now an empty `ActionColumn` can be generated with a flag for the Column Manager. The `Cell` prop can be overridden by the user if they want to populate the column or not (see [here](https://github.com/iTwin/iTwinUI-react/pull/564/files#diff-7e8181f0c561b1370b6fadd31d714fcf9293c0fd45d8de3f75d055f553d922a8R2439) in Table test)

https://github.com/iTwin/iTwinUI-react/pull/564/files#diff-7e8181f0c561b1370b6fadd31d714fcf9293c0fd45d8de3f75d055f553d922a8R2439